tableauserverclient/models/interval_item.py

```diff
@@ -246,21 +246,35 @@ def interval(self):
 
     @interval.setter
     def interval(self, interval_values):
-        # This is weird because the value could be a str or an int
-        # The only valid str is 'LastDay' so we check that first. If that's not it
-        # try to convert it to an int, if that fails because it's an incorrect string
-        # like 'badstring' we catch and re-raise. Otherwise we convert to int and check
-        # that it's in range 1-31
+        # Valid monthly intervals strings can contain any of the following
+        #   day numbers (1-31)
+        #   relative day within the month (First, Second, ... Last)
+        #   week days (Sunday, Monday, ... LastDay)
+        VALID_INTERVALS = [
+            "Sunday",
+            "Monday",
+            "Tuesday",
+            "Wednesday",
+            "Thursday",
+            "Friday",
+            "Saturday",
+            "LastDay",
+            "First",
+            "Second",
+            "Third",
+            "Fourth",
+            "Fifth",
+            "Last",
+        ]
         for interval_value in interval_values:
-            error = "Invalid interval value for a monthly frequency: {}.".format(interval_value)
-
-            if interval_value != "LastDay":
-                try:
-                    if not (1 <= int(interval_value) <= 31):
-                        raise ValueError(error)
-                except ValueError:
-                    if interval_value != "LastDay":
-                        raise ValueError(error)
+            try:
+                if not (1 <= int(interval_value) <= 31):
+                    error = f"Invalid monthly numeric frequency interval: {interval_value}."
```

Comment:
This error message can never happen right? Its ValueError will get caught below and either swallowed or a new ValueError will be thrown. I haven't used Python much, is it common to use exceptions to control flow?

Maybe use the `isinstance(interval, (int, float))` construct like on line 80 to detect if it's a number first.

Reply:
Well in theory I think it would fail right there if the numerical value was 32 or "32".

I agree using exceptions here is a little funky. I was trying to fit in with the methods used already.

Reply:
Using exceptions for control flow isn't uncommon in python. The `except ValueError` on 274 would indeed catch and consume the one raised on 273. The type check proposed by @anyoung-tableau does seem like a good alternative to let the different errors surface.

Reply:
I'll take another look at it. The challenge is that even numbers are strings by the time they land here.

Reply:
Easiest way to do it seems like explicitly listing the known allowed digits: 
for value in range(1, 32):
    VALID_INTERVALS.add(str(value))